bpf/lib/maps.h

```diff
@@ -186,21 +186,23 @@ struct bpf_elf_map __section_maps ENCRYPT_MAP = {
 	.max_elem	= 1,
 };
 
-struct egress_key {
-	struct bpf_lpm_trie_key lpm_key;
-	__u32 sip;
-	__u32 dip;
-};
-
 #ifdef ENABLE_EGRESS_GATEWAY
-struct bpf_elf_map __section_maps EGRESS_MAP = {
+struct bpf_elf_map __section_maps EGRESS_POLICY_MAP = {
 	.type		= LPM_MAP_TYPE,
-	.size_key	= sizeof(struct egress_key),
-	.size_value	= sizeof(struct egress_info),
+	.size_key	= sizeof(struct egress_policy_key),
+	.size_value	= sizeof(struct egress_policy),
 	.pinning	= PIN_GLOBAL_NS,
-	.max_elem	= EGRESS_MAP_SIZE,
+	.max_elem	= EGRESS_POLICY_MAP_SIZE,
 	.flags		= BPF_F_NO_PREALLOC,
 };
+
+struct bpf_elf_map __section_maps EGRESS_CT_MAP = {
+	.type		= BPF_MAP_TYPE_LRU_HASH,
```

Comment:
Correct, if the map is full and a connection gets evicted, it will likely be moved to a different gateway node (at least with the current selection strategy based on `random()`) and it will get reset because of the lack of SNAT state.
At the same time I don't think there's much we can do about it :thinking: having a log in the agent could be useful, but I'm not sure how to properly detect when an element gets evicted (maybe we can just periodically check if the number of elements in the map is close to its size?)

Reply:
Seems it's also the case for the usual CT map. So let's keep it as is for now :slightly_smiling_face: